py/templates/users/login.html

```diff
@@ -37,7 +39,10 @@ <h3>first time here?</h3>
     var form = $('#login-form');
     
     form.submit(function(e) {
-        if (!usernameIsValid() || !passwordIsValid() || !emailIsValid()) {
+        var check1 = !usernameIsValid();
```

Comment:
Funny change :) Probably it is a debug remnant and I'll merge it again into single line if you don't mind :)

Reply:
No, it's intended. Since in javascript especially in 'or' checking, once the earlier left hand side of 'or' resolved to be true, the right hand side won't be checked any more. Thus, it would only shows warning correlated to the left hand side of 'or' checking. By separating into each individual line, we 'forced' the checking.